receiver/statsdreceiver/receiver.go

```diff
@@ -69,12 +69,13 @@ func New(
 
 func buildTransportServer(config Config) (transport.Server, error) {
 	// TODO: Add TCP/unix socket transport implementations
-	switch strings.ToLower(config.NetAddr.Transport) {
-	case "", "udp":
-		return transport.NewUDPServer(config.NetAddr.Endpoint)
+	trans, err := transport.NewTransport(strings.ToLower(config.NetAddr.Transport))
+	switch trans {
+	case transport.UDP, transport.UDP4, transport.UDP6:
+		return transport.NewUDPServer(trans, config.NetAddr.Endpoint)
 	}
 
-	return nil, fmt.Errorf("unsupported transport %q", config.NetAddr.Transport)
+	return nil, fmt.Errorf("error building server with transport %q: %w", config.NetAddr.Transport, err)
```

Comment:
usually we try to avoid to "error" work in the error message to avoid when printing it something like:

```
error when bootstraping: error when opening connection: error building server
```

Reply:
This error has been removed and does not exist anymore.